src/ess/reduce/scripts/grow_nexus.py

```diff
@@ -0,0 +1,119 @@
+import argparse
+
+import h5py
+
+
+def grow_nexus_file(file, detector_scale, monitor_scale):
+    for group in file.values():
+        if group.attrs.get('NX_class', '') == 'NXentry':
+            entry = group
+            break
+    for group in entry.values():
+        if group.attrs.get('NX_class', '') == 'NXinstrument':
+            instrument = group
+            break
+    for group in instrument.values():
+        if (nx_class := group.attrs.get('NX_class', '')) in (
+            'NXdetector',
+            'NXmonitor',
+        ):
+            for subgroup in group.values():
+                if subgroup.attrs.get('NX_class', '') == 'NXevent_data':
+                    _scale_group(
+                        subgroup,
+                        scale=detector_scale
+                        if nx_class == 'NXdetector'
+                        else monitor_scale,
+                    )
+
+
+def _scale_group(event_data: h5py.Group, scale: int | float):
+    event_index = (event_data['event_index'][:] * scale).round().astype('uint')
+    event_data['event_index'][:] = event_index
+
+    size = event_data['event_id'].size
+    event_data['event_id'].resize(event_index[-1], axis=0)
+    event_data['event_time_offset'].resize(event_index[-1], axis=0)
+
+    for s in range(max(int(scale - 1), 0)):
+        event_data['event_id'][s * size : (s + 1) * size] = event_data['event_id'][
+            :size
+        ]
+        event_data['event_time_offset'][s * size : (s + 1) * size] = event_data[
+            'event_time_offset'
+        ][:size]
+
+    last = max(int(scale - 1), 0) * size
+    chunksize = event_data['event_id'].size - last
+    event_data['event_id'][last:] = event_data['event_id'][:chunksize]
+    event_data['event_time_offset'][last:] = event_data['event_time_offset'][:chunksize]
+
+
+def _parse_args(args):
+    if args.detector_scale < 1:
+        print(
+            (
+                'Warning: Reducing event count, but chunking is left untouched. '
+                'If the chunk size is large the resulting file may actually not shrink. '  # noqa: E501
+                'If that is the case, use something like '
+                '`h5repack -i {0}.nxs -o {0}-rechunked.nxs -l CHUNK=N` '
+                'to produce a smaller file.'.format(args.filename.rpartition('.')[0])
+            )
+        )
+
+    detector_scale = (
+        args.detector_scale if args.detector_scale < 1 else round(args.detector_scale)
+    )
+    monitor_scale = (
+        round(args.monitor_scale)
```

Comment:
This would be in disagreement with the detector scale if a `monitor_scale = 0.5` is given?
Again, probably removing the ability to scale with < 1 would help?